scripts/validate_wheel.sh

```diff
@@ -0,0 +1,166 @@
+while IFS= read -r line; do
+    if [ -n "$(echo $line | tr -d '[:space:]')" ]; then
+        eval "$line"
+    fi
+done <<< "$ompi_script"
+
+# Run core tests
+echo "Running core tests."
+python3 -m pip install pytest numpy
+python3 -m pytest -v "$root_folder/tests" \
+    --ignore "$root_folder/tests/backends" \
+    --ignore "$root_folder/tests/parallel" \
+    --ignore "$root_folder/tests/domains"
+if [ ! $? -eq 0 ]; then
+    echo -e "\e[01;31mPython tests failed.\e[0m" >&2
+    status_sum=$((status_sum+1))
+fi
+
+# Run backend tests
+echo "Running backend tests."
+python3 -m pip install --user fastapi uvicorn llvmlite
+for backendTest in "$root_folder/tests/backends"/*.py; do 
+    python3 -m pytest -v $backendTest
+    # Exit code 5 indicates that no tests were collected,
+    # i.e. all tests in this file were skipped, which is the case
+    # for the mock server tests since they are not included.
+    status=$?
+    if [ ! $status -eq 0 ] && [ ! $status -eq 5 ]; then
+        echo -e "\e[01;31mPython backend test $backendTest failed with code $status.\e[0m" >&2
+        status_sum=$((status_sum+1))
+    fi
+done
+
+# Run platform tests
+echo "Running platform tests."
+for parallelTest in "$root_folder/tests/parallel"/*.py; do 
+    python3 -m pytest -v $parallelTest
+    if [ ! $? -eq 0 ]; then
+        echo -e "\e[01;31mPython platform test $parallelTest failed.\e[0m" >&2
+        status_sum=$((status_sum+1))
+    fi
+done
+
+# Run examples
+status_sum=0
+for ex in `find "$root_folder/examples" -name '*.py' -not -path '*/providers/*'`; do
+    python3 "$ex"
+    if [ ! $? -eq 0 ]; then
+        echo -e "\e[01;31mFailed to execute $ex.\e[0m" >&2
+        status_sum=$((status_sum+1))
+    fi
+done
+
+# Run remote-mqpu platform test
+cudaq_location=`python3 -m pip show cuda-quantum | grep -e 'Location: .*$'`
+qpud_exe="${cudaq_location#Location: }/bin/cudaq-qpud"
+if [ -x "$(command -v nvidia-smi)" ]; 
+then nr_gpus=`nvidia-smi --list-gpus | wc -l`
+else nr_gpus=0
+fi
+server1_devices=`echo $(seq $((nr_gpus >> 1)) $((nr_gpus - 1))) | tr ' ' ,`
+server2_devices=`echo $(seq 0 $((($nr_gpus >> 1) - 1))) | tr ' ' ,`
+echo "Launching server 1..."
+servers="localhost:12001"
+CUDA_VISIBLE_DEVICES=$server1_devices mpiexec --allow-run-as-root -np 2 "$qpud_exe" --port 12001 &
+if [ -n "$server2_devices" ]; then
+    echo "Launching server 2..."
+    servers+=",localhost:12002"
+    CUDA_VISIBLE_DEVICES=$server2_devices mpiexec --allow-run-as-root -np 2 "$qpud_exe" --port 12002 &
+fi
+
+python3 "$root_folder/snippets/using/cudaq/platform/sample_async_remote.py" \
+    --backend nvidia-mgpu --servers "$servers"
+if [ ! $? -eq 0 ]; then
+    echo -e "\e[01;31mRemote platform test failed.\e[0m" >&2
+    status_sum=$((status_sum+1))
+fi
+kill %1 && wait %1 2> /dev/null
+if [ -n "$server2_devices" ]; then
+    kill %2 && wait %2 2> /dev/null
```

Comment:
I was seeing segfaults reported with `kill` and adding `2> /dev/null` did not make them go away (since the errors were coming from the original processes). They went away for me with `kill -9`. If you are seeing the same thing, I'd recommend adding the `-9` in order to make the logs not have benign segfaults.

Reply:
I intentionally did not use the -9 because I wanted to see the segfaults on tear down. While they are potentially benign, they still indicate an issue during tear down (which Thien has since fixed).